dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -130,18 +134,39 @@ internal class DatadogRumMonitor(
             devLogger.w("User action $action was ignored: previous action still active.")
             return
         }
-        sendUnclosedAction()
-        val nanoTime = System.nanoTime()
-        val eventData = RumEventData.UserAction(
-            action,
-            UUID.randomUUID(),
-            nanoTime
-        )
+        registerUserAction(action, attributes)
+    }
 
-        val event = rumEvent(eventData, attributes)
-        lastActionRelatedEventNs.set(nanoTime)
-        activeActionEvent = event
-        activeActionData = eventData
+    @Synchronized
+    internal fun startUserAction() {
```

Comment:
What if we need to pass a name or some attributes at the beginning of the action ?